### PR TITLE
Django2

### PR DESCRIPTION
**Pull request by sheagcraig**
This pull request updates a number of minor things that are API changes between django 1.10 and django 1.11+ (we're going straight to the current release, which is django 2.1.4).

At this time, I also updated a number of other dependency packages to ensure we got versions that were compatible with newer django versions.

The changes are mostly summarized thusly:
- All `ForeignKey` relationships now require an explicit `on_delete` parameter. They have all been set to `models.CASCADE` which is the existing behavior; I have put in a feature request to examine some of the special cases where we might not want this to happen (mostly involving retaining machine records). This change accounts for most of the diff (it had to be added to all FK references in the migrations too).
- Some methods have moved; `reverse` is one.
- `User.is_authenticated` is now an attribute.
- The settings `MIDDLEWARE_CLASSES` has been renamed to `MIDDLEWARE`.
- Updated custom middleware to inherit the helper/transitional mixin so that they conform to the new API.

### CODE REVIEW

**Comment by sheagcraig** [comment]
I'm noticing that no Munki items show up in the machine detail page right now. Looking into it...

**Comment by sheagcraig** [comment]
...solved.

File: server/non_ui_views.py [hunk 1]
```diff
@@ -250,17 +250,25 @@ def checkin(request):
...
-    machine.report = report_bytes
+    # the bytes as unicode (otherwise it gets munged).
+    machine.report = report_bytes.decode()
     machine.console_user = get_console_user(report_data)
 
+    # We need to save now or else further processing of related fields
+    # will fail.
+    try:
+        machine.save()
+    except ValueError:
```
Comment (grahamgilbert) [inline server/non_ui_views.py]: If failing here means that nothing will be processed, should we bother continuing or just bail?

Comment (sheagcraig) [reply server/non_ui_views.py]: Probably just bail. I didn't want it to throw an exception; but we should probably return an error then.

**Pull request merged**